lib/js/components/Pagination/Pagination.vue

```diff
@@ -0,0 +1,369 @@
+const FIRST_PAGE_NUMBER = 1;
+
+export default {
+	name: 'Pagination',
+	components: { IconButton },
+	props: {
+		layout: {
+			type: String,
+			default: PAGINATION_LAYOUTS.DEFAULT,
+			validate(layout) {
+				return Object.values(PAGINATION_LAYOUTS).includes(layout);
+			},
+		},
+		currentPage: {
+			type: Number,
+			default: FIRST_PAGE_NUMBER,
+			validate(page) {
+				return page > 0;
+			},
+		},
+		itemsPerPage: {
+			type: Number,
+			default: PAGINATION_DEFAULT_ITEMS_PER_PAGE,
+		},
+		initialPage: {
+			type: Number,
+			default: FIRST_PAGE_NUMBER,
+		},
+		isCentered: {
+			type: Boolean,
+			default: false,
+		},
+		itemsTotalAmount: {
+			type: Number,
+			required: true,
+		},
+	},
+	data() {
+		return {
+			PAGINATION_LAYOUTS: Object.freeze(PAGINATION_LAYOUTS),
+			ICON_BUTTON_SIZES: Object.freeze(ICON_BUTTON_SIZES),
+			ICON_BUTTON_COLORS: Object.freeze(ICON_BUTTON_COLORS),
+			ICON_BUTTON_STATES: Object.freeze(ICON_BUTTON_STATES),
+			ICONS: Object.freeze(ICONS),
+		};
+	},
+	computed: {
+		lastPage() {
+			return Math.ceil(this.itemsTotalAmount / this.itemsPerPage);
+		},
+		page() {
+			if (this.pageIsSmallerThanFirstPage(this.currentPage)) {
+				return FIRST_PAGE_NUMBER;
+			}
+
+			if (this.pageIsLargerThanLastPage(this.currentPage)) {
+				return this.lastPage;
+			}
+
+			return this.currentPage;
+		},
+		items() {
+			let delta: number;
+			if (this.lastPage <= MAX_DISPLAYED_ITEMS) {
+				// delta === 7: [1 2 3 4 5 6 7]
+				delta = 7;
+			} else {
+				// delta === 2: [1 ... 4 5 6 ... 10]
+				// delta === 4: [1 2 3 4 5 ... 10]
+				delta = this.currentPage > 4 && this.currentPage < this.lastPage - 3 ? 2 : 4;
+			}
+
+			const range = {
+				start: Math.round(this.currentPage - delta / 2),
+				end: Math.round(this.currentPage + delta / 2),
+			};
+
+			if (range.start - 1 === 1 || range.end + 1 === this.lastPage) {
+				range.start += 1;
+				range.end += 1;
+			}
+
+			let pages: any =
+				this.currentPage > delta
+					? this.getRange(
+							Math.min(range.start, this.lastPage - delta),
+							Math.min(range.end, this.lastPage),
+					  )
+					: this.getRange(1, Math.min(this.lastPage, delta + 1));
+
+			const withDots = (value, pair) => (pages.length + 1 !== this.lastPage ? pair : [value]);
+
+			if (pages[0] !== 1) {
+				pages = withDots(1, [1, ELLIPSIS_FILL]).concat(pages);
+			}
+
+			if (pages[pages.length - 1] < this.lastPage) {
+				pages = pages.concat(withDots(this.lastPage, [ELLIPSIS_FILL, this.lastPage]));
+			}
+
+			return pages;
+		},
+	},
+	methods: {
+		getRange(start: number, end: number) {
+			return Array(end - start + 1)
+				.fill(null)
+				.map((_v, i) => i + start);
+		},
+		async changePage(page) {
+			if (this.page === page) {
+				return;
+			}
+
+			this.$emit('change-page', page);
+		},
+		isPage(item) {
+			return typeof item === 'number';
+		},
+		pageIsSmallerThanFirstPage(page) {
+			return page < FIRST_PAGE_NUMBER || page % FIRST_PAGE_NUMBER !== 0;
```

Comment:
Is the second part just to sanitize floats? If so, it's hidden and not obvious at all. Let's validate it in prop validator, not here. If there is a different reason to have it here, I don't see it.